osu.Game/Tests/Beatmaps/TestWorkingBeatmap.cs

```diff
@@ -1,24 +1,46 @@
 ﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
+using System.Linq;
 using osu.Framework.Audio.Track;
 using osu.Framework.Graphics.Textures;
 using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Objects.Types;
 
 namespace osu.Game.Tests.Beatmaps
 {
     public class TestWorkingBeatmap : WorkingBeatmap
     {
+        public TestWorkingBeatmap(RulesetInfo ruleset)
+            : this(new TestBeatmap(ruleset))
+        {
+        }
+
         public TestWorkingBeatmap(Beatmap beatmap)
             : base(beatmap.BeatmapInfo)
         {
             this.beatmap = beatmap;
         }
 
         private readonly Beatmap beatmap;
-
         protected override Beatmap GetBeatmap() => beatmap;
         protected override Texture GetBackground() => null;
-        protected override Track GetTrack() => new TrackVirtual();
+
+        protected override Track GetTrack()
+        {
+            var lastObject = beatmap.HitObjects.LastOrDefault();
+            if (lastObject != null)
+                return new TestTrack(((lastObject as IHasEndTime)?.EndTime ?? lastObject.StartTime) + 1000);
+            return new TrackVirtual();
+        }
+
+        private class TestTrack : TrackVirtual

```

Comment:
planning on adding more to this class? if not, could possibly just set the length directly

Reply:
Length setter is protected (probably for a good reason). That's the only reason this class exists.

Reply:
hmm, interesting. i get the feeling we'll need to unprotect at some point, but this is fine for now i guess!